frontend/lib/resolution/prims.cpp

```diff
@@ -511,6 +526,255 @@ static QualifiedType primObjectToInt(Context* context, const CallInfo& ci) {
   return QualifiedType(argType.kind(), IntType::get(context, 64));
 }
 
+// TODO: What should be done for 'MAYBE_GENERIC', if anything?
+static QualifiedType primIsGenericType(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 1) return {};
+  bool eval = false;
+  auto qt = ci.actual(0).type();
+  if (auto t = qt.type()) {
+    auto g = t->genericity();
+
+    if (g == Type::MAYBE_GENERIC) {
+      QualifiedType qtEval = qt;
+      auto ct = t->toCompositeType();
+      if (auto cls = t->toClassType()) ct = cls->basicClassType();
+
+      // A generic type with defaults will be computed to have default subs
+      // at this point, so calling 'getTypeGenericity' on it will produce
+      // 'CONCRETE' instead of 'GENERIC_WITH_DEFAULTS'. Back up and use the
+      // uninstantiated type to produce the latter.
+      if (ct) {
+        if (auto ctGeneric = ct->instantiatedFromCompositeType()) {
+          qtEval = QualifiedType(qt.kind(), ctGeneric);
+        }
+      }
+      g = getTypeGenericity(context, qtEval);
+    }
+
+    // Both cases are considered 'generic' for this primitive.
+    eval = (g == Type::GENERIC || g == Type::GENERIC_WITH_DEFAULTS);
+  }
+  return makeParamBool(context, eval);
+}
+
+// TODO: No to 'FLAG_C_PTR_CLASS', 'FLAG_DATA_CLASS', 'FLAG_REF'.
+static QualifiedType primIsClassType(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 1) return {};
+  bool eval = false;
+  if (auto t = ci.actual(0).type().type()) {
+    // TODO: Is it possible for the second and third to appear at all?
+    eval = t->isClassType()      ||  // E.g., 'owned C'.
+           t->isBasicClassType() ||  // E.g., 'C' in 'owned C'.
+           t->isAnyClassType();      // E.g., 'owned class'.
```

Comment:
I'm not sure, Anna asked the same question and my guess is that it would return false, because it is kinda WYSIWG. But I think that no one should be asking about `_owned` except for the implementation, anyhow, so that shouldn't come up too much in practice.